access-common/src/main/java/edu/unc/lib/dl/ui/service/LorisContentService.java

```diff
@@ -234,25 +236,24 @@ public String getManifest(HttpServletRequest request, List<BriefObjectMetadata>
         return iiifMapper.writeValueAsString(manifest.addSequence(seq));
     }
 
-    private String Jp2Pid(List<Datastream> datastream) {
+    private String jp2Pid(Datastream datastream) {
         if (datastream != null) {
-            for (Datastream stream : datastream) {
-                String streamValues = stream.toString();
-                if (streamValues.trim().startsWith("jp2")) {
-                    String[] uuid_parts = streamValues.split("\\|");
-                    if (uuid_parts.length == 7) {
-                        return uuid_parts[6];
-                    }
-                    return uuid_parts[2].split("\\.")[0];
+            String streamValues = datastream.toString();
```

Comment:
the Datastream class already explodes the string representation of the solr field, so you can just ask it for whatever fields you're trying to get. `owner` is the 6th field, etc. See `edu.unc.lib.dl.search.solr.model.Datastream.Datastream(String)`